cli/githubutil.py

```diff
@@ -156,18 +157,17 @@ def release_and_prepare_next_dev_cycle(
         )
 
     if should_generate_release_notes:
-        release_notes_md = rls_notes_as_markdown_and_post_to_slack(
+        release_notes = fetch_release_notes(
             github_repository_owner=github_repository_owner,
             github_repository_name=github_repository_name,
             github_cfg=github_cfg,
             repo_dir=repo_dir,
             github_helper=helper,
             repository_branch=repository_branch,
-            slack_cfg_name=slack_cfg_name,
-            slack_channel=slack_channel,
-            release_version=release_version
         )
+        release_notes_md = release_notes.to_markdown()
```

Comment:
would it not suffice to store the release_notes object (or None) and convert it to a str where required?

Also, please initialise variables with default values before conditionally assigning them with values (and keep the amount of those variables to a minimum).